addons/hr_expense/models/hr_expense.py

```diff
@@ -454,8 +458,28 @@ def write(self, vals):
                     self.sheet_id.write({'employee_id': vals['employee_id']})
                 elif len(employees) > 1:
                     self.sheet_id = False
+        if 'sheet_id' in vals:
+            for expense in self:
+                prev_sheet = expense_to_prev_sheet[expense]
+                checksums = set((expense.attachment_ids - prev_sheet.expense_line_ids.attachment_ids).mapped('checksum'))
+                prev_sheet.attachment_ids.filtered(lambda att: att.checksum in checksums).with_context(sync_attachment=False).unlink()
+                if vals['sheet_id']:
+                    if expense.sheet_id != prev_sheet:
+                        for attachment in expense.attachment_ids.with_context(sync_attachment=False):
+                            attachment.copy({
+                                'res_model': 'hr.expense.sheet',
+                                'res_id': vals['sheet_id'],
+                            })
```

Comment:
A comment for this one would be nice :)
Also, nitpick but "prev" instead of "previous" isn't saving a lot of characters and less clear imo

Reply:
I've added a few comments to explain, it's true it's the most tricky part of the patch, let me know if something isn't clear.
Also renamed prev -> previous, since it's true that prev could be misinterpreted as "preview" for ex.